src/Hazel.hs

```diff
@@ -352,19 +344,15 @@ check dirs ctx ty tm = case tm of
       "[check Primop] primop (" ++ show p ++ ") type mismatch"
     return ctx
 
-  Let pat letTm vTm -> do
-    (leftovers, tmTy) <- infer (Let1:dirs) ctx letTm
-    patternTy <- typePattern pat tmTy
-    -- XXX do we need to reverse these?
-    let bodyCtx = patternTy ++ leftovers
-        arity = length patternTy
-    newCtx <- check (Let2:dirs) bodyCtx ty vTm
-
-    -- Check that the body consumed all the arguments
-    let (bodyUsage, leftovers2) = splitAt arity newCtx
-    forM_ bodyUsage $ \(_ty, usage) ->
-      assert (usage /= UseOnce) dirs
-        "[check Let] must consume linear bound variables"
+  Let rhsTm vTm -> do
+    (leftovers, rhsTy) <- infer (Let1:dirs) ctx rhsTm
+    -- FIXME: the real usage should flow from inference, whereas we currently
+    -- only have the pretype:
+    let fakeIncorrectUsage = UseOnce
+        bodyCtx = (rhsTy, fakeIncorrectUsage):leftovers
+    leftovers2@((_, usedRhs):_) <- check (Let2:dirs) bodyCtx ty vTm
```

Comment:
Shouldn't this be `(_, usedRhs):leftovers2` since we don't want the type / usage of the bound variable to escape?